### PR TITLE
DP-7592:  MainNav molecule component.

### PR DESCRIPTION
**Pull request by smurrayatwork**
**Description:** 
This adds the Main Nav molecule to the  storybook, as well as support for adding its markdown info for display. There is still some functionality from mayflower's javascript that should be implemented in the future: keypress support, menu animations, and window resize events.

**Jira:** 
https://jira.state.ma.us/browse/DP-7592

**To Test:**
- [ ] Pull the branch and run 'npm run storybook'.
- [ ] Browse to http://localhost:6006/ (or the port that you chose instead) and verify there's now an MainNav link on the left side under Molecules. Click on it and verify that the molecule displayed matches the attached screenshot.
- [ ] Verify that each knob changes the corresponding component.

**Post Deploy Steps:**
N/A

**Relevant Screenshots/GIFs:**
screen shot 2018-02-14 at 10 40 49 am

**Additional Notes:**
N/A
---

[Peer Review Checklist](docs/Peer%20review%20checklist.md)

### CODE REVIEW

File: src/components/molecules/MainNav/MainNav.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+### Description
```
Comment (mrjmd) [inline src/components/molecules/MainNav/MainNav.md]: I tried to figure out why the ^^ package-lock diff was showing up, but with node 8.9.4 and npm 5.6.0, after deleting node_modules and package-lock and npm installing again, it shows up for me too in this branch (but not in latest develop). Weird.

File: src/components/molecules/MainNav/MainNav.stories.js [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+import { withInfo } from '@storybook/addon-info';
+import { withKnobs, text, boolean, object } from '@storybook/addon-knobs/react';
+
+import MainNav from './index';
+import MainNavDocs from './MainNav.md';
+
+storiesOf('molecules', module).addDecorator(withKnobs)
+  .add('MainNav', withInfo(`<div>${MainNavDocs}</div>`)(() => {
+    const props = {
+      mainNav: [
```
Comment (mrjmd) [inline src/components/molecules/MainNav/MainNav.stories.js]: Should we extract mainNav here to a separate JSON file?

File: src/components/molecules/MainNav/index.js [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+    this.setState({
+      navSelected: -1
+    });
+  }
+  render() {
+    return(
+      <section className="ma__main-nav">
+        <ul className="ma__main-nav__items js-main-nav">
+          {this.props.mainNav.map((item, index) => {
+  const buttonId = `button${index}`;
```
Comment (mrjmd) [inline src/components/molecules/MainNav/index.js]: All this spacing looks borked.

File: src/components/molecules/MainNav/MainNav.md (hunk 1 above)
Comment (mrjmd) [reply src/components/molecules/MainNav/MainNav.md]: Nm we fixed it :)

File: src/components/molecules/MainNav/index.js [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+        </ul>
+      </section>
+    );
+  }
+}
+
+MainNav.propTypes = {
+  /** An array of navigation objects to display in the main nav */
+  mainNav: PropTypes.arrayOf(PropTypes.shape({
+    href: PropTypes.string.isRequired,
```
Comment (avrilpearl) [inline src/components/molecules/MainNav/index.js]: i don't think you need this sub element - `mainNav` here.

File: src/components/molecules/MainNav/index.js [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+      navSelected: e.currentTarget.id
+    });
+  }
+  mouseOut() {
+    this.setState({
+      navSelected: -1
+    });
+  }
+  render() {
+    return(
```
Comment (avrilpearl) [inline src/components/molecules/MainNav/index.js]: here, when you remove mainNav below in the prop definition
you should have `const mainNav = this.props`

**Pull request merged**